apps/settings/js/modules/addon_manager.js

```diff
@@ -105,6 +108,108 @@ define(function(require) {
     });
   }
 
+  function renameAddon(addon, name) {
+    if (!name) {
+      return Promise.reject('no name given');
+    }
+
+    if (!isAddon(addon)) {
+      return Promise.reject('not an addon');
+    }
+
+    var manifest = getManifest(addon);
+    if (name === manifest.name) {
+      return Promise.reject('name is unchanged');
+    }
+
+    // To rename an addon we are creating a new DOMApplication that will have
+    // an updated name. Other manifest attributes are preserved.
+    return new Promise(function(resolve, reject) {
+      // Export an addon as a blob.
+      addon.export().then(function(blob) {
+        // Copy over manifest object.
+        var newManifest = Object.assign({}, manifest);
+        newManifest.name = name;
+
+        deleteAddon(addon).then(function() {
+          blobToArrayBuffer(blob).then(function(arrayBuffer) {
+            var addonBlob = generate(newManifest, unpackScript(arrayBuffer));
+            install(addonBlob).then(resolve).catch(reject);
+          }).catch(reject);
+        }).catch(reject);
+      }).catch(reject);
```

Comment:
Catching the rejections seems redundant if we don't do further handling and rejects directly. If this is intended, a catch function at the end of the promise chain should be sufficient.